### PR TITLE
qa: switch to use the merge fragment for fscrypt

### PR DESCRIPTION
**Pull request by lxbsz**
PR https://github.com/ceph/ceph/pull/48183 have add the merge fragment support. This will switch to use fragment to simplify the code.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by lxbsz** [comment]
All the [fscrypt qa tests](https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/) are running.

**Comment by lxbsz** [comment]
Only updated the commit comments.

File: qa/suites/fs/fscrypt/tasks/0-client.yaml
Comment (dparmar18) [inline qa/suites/fs/fscrypt/tasks/0-client.yaml:5]: we will have to keep track on this right whether and when this feature is added to distro kernels and/or ceph-fuse?

Comment (lxbsz) [reply qa/suites/fs/fscrypt/tasks/0-client.yaml:5]: I added this comment just as Patrick mentioned in the previous PR, just to make it clear why we add this restriction here and in which case we can remove it.

Comment (dparmar18) [reply qa/suites/fs/fscrypt/tasks/0-client.yaml:5]: right, i mean once the distro kernels/ceph-fuse are capable of fscrypt then we need to change it here and also here https://github.com/ceph/ceph/pull/48183/files#diff-63be9c532a24d81b6391bb01e3d994eef8eb6a948154c227164063e6acd592f7R3-R7. So it would be better that we note it down somewhere, or better way is to link these two in the fscrypt feature PR

Comment (lxbsz) [reply qa/suites/fs/fscrypt/tasks/0-client.yaml:5]: I just linked this to the fscrypt feature PR https://github.com/ceph/ceph/pull/50728. Thanks.

**Comment by dparmar18** [comment]
there's one job running close to 5 hours - https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/7224905/, I see a lot of failures in generic/XXX that are not visible in other jobs.

**Comment by lxbsz** [comment]
> there's one job running close to 5 hours - https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/7224905/, I see a lot of failures in generic/XXX that are not visible in other jobs.

Yeah, I have checked it. It's because the `osd6` dead for some reasons. I need to wait the job to be aborted and check the osd logs.

**Comment by lxbsz** [comment]
I ran the [qa tests](https://pulpito.ceph.com/xiubli-2023-03-29_10:29:15-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/) again. This time the `fscrypt-commom` test failed, but it failed from:

```
generic/020       - output mismatch (see /tmp/tmp.jrgZljkscrxfstests-dev/results//generic/020.out.bad)
    --- tests/generic/020.out   2023-03-29 10:57:23.284330119 +0000
    +++ /tmp/tmp.jrgZljkscrxfstests-dev/results//generic/020.out.bad    2023-03-29 11:08:24.814914036 +0000
    @@ -47,9 +47,13 @@
     user.snrub="fish2\012"

     *** really long value
    -0000000 00
    -*
    -ATTRSIZE
    +attr_set: No space left on device
    +Could not set "long_attr" for <TESTFILE>
    +attr_get: No data available
    +Could not get "long_attr" for <TESTFILE>
    +0000000
    +attr_remove: No data available
    +Could not remove "long_attr" for <TESTFILE>
     *** set/get/remove really long names (expect failure)
     attr_set: Invalid argument
     Could not set "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX" for <TESTFILE>
```

This is because in `Ubuntu` it doesn't support the `selinux`:
```
2023-03-29T10:44:58.481 INFO:teuthology.run_tasks:Running task selinux...
2023-03-29T10:44:58.530 DEBUG:teuthology.task.selinux:Excluding smithi043: OS 'ubuntu' does not support SELinux
2023-03-29T10:44:58.530 DEBUG:teuthology.task.selinux:Excluding smithi161: OS 'ubuntu' does not support SELinux
2023-03-29T10:44:58.530 DEBUG:teuthology.task.selinux:Excluding smithi174: OS 'ubuntu' does not support SELinux
2023-03-29T10:44:58.531 DEBUG:teuthology.task.selinux:Getting current SELinux state
2023-03-29T10:44:58.531 DEBUG:teuthology.task.selinux:Existing SELinux modes: {}
```

This is a [known bug](https://patchwork.kernel.org/project/ceph-devel/patch/20230319062928.278235-1-xiubli@redhat.com/), and have been in the `xfstests-dev` [for-next](https://github.com/kdave/xfstests/commit/2d7d5f49e7276a8f40ca28d03408a4d807be8936) branch.

**Comment by lxbsz** [comment]
> > there's one job running close to 5 hours - https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/7224905/, I see a lot of failures in generic/XXX that are not visible in other jobs.
> 
> Yeah, I have checked it. It's because the `osd6` dead for some reasons. I need to wait the job to be aborted and check the osd logs:
> 
> ```
> [ 2045.692294] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> [ 2045.699532] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> [ 2045.706782] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> [ 2045.714312] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> [ 2045.721540] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> [ 2045.728845] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> ```

This is the same issue with https://github.com/ceph/ceph/pull/50728#issuecomment-1488610649.

**Review by rishabh-d-dave — COMMENTED** [review]
LGTM. I'll this a second look and approve if after some time.

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by lxbsz** [comment]
> > > > there's one job running close to 5 hours - https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/7224905/, I see a lot of failures in generic/XXX that are not visible in other jobs.
> > > 
> > > 
> > > Yeah, I have checked it. It's because the `osd6` dead for some reasons. I need to wait the job to be aborted and check the osd logs:
> > > ```
> > > [ 2045.692294] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > [ 2045.699532] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > [ 2045.706782] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > [ 2045.714312] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > [ 2045.721540] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > [ 2045.728845] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > ```
> > 
> > 
> > This is the same issue with [#50728 (comment)](https://github.com/ceph/ceph/pull/50728#issuecomment-1488610649).
> 
> okay so once your fix has been merged, it should go fine right

Yeah, once that fix is applied to the `xfstests-dev` repo's master branch it will be fetched to `git.ceph.com` and then this test failure will disappear.

**Comment by dparmar18** [comment]
> > > > > > there's one job running close to 5 hours - https://pulpito.ceph.com/xiubli-2023-03-29_05:17:18-fs:fscrypt-wip-lxb-20230316-2212-unlink-revert-distro-default-smithi/7224905/, I see a lot of failures in generic/XXX that are not visible in other jobs.
> > > > > 
> > > > > 
> > > > > Yeah, I have checked it. It's because the `osd6` dead for some reasons. I need to wait the job to be aborted and check the osd logs:
> > > > > ```
> > > > > [ 2045.692294] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > [ 2045.699532] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > [ 2045.706782] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > [ 2045.714312] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > [ 2045.721540] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > [ 2045.728845] libceph: osd6 (1)172.21.15.181:6825 socket error on read^M
> > > > > ```
> > > > 
> > > > 
> > > > This is the same issue with [#50728 (comment)](https://github.com/ceph/ceph/pull/50728#issuecomment-1488610649).
> > > 
> > > 
> > > okay so once your fix has been merged, it should go fine right
> > 
> > 
> > Yeah, once that fix is applied to the `xfstests-dev` repo's master branch it will be fetched to `git.ceph.com` and then this test failure will disappear.
> 
> Or if the job is scheduled with the `rhel/Centos` OS the test will pass too. Only will it fail in `Ubuntu`.

okay but when running full suite, it will show failures for `Ubuntu`, so is it feasible to merge this or wait till your fix is merged in `xfstests-dev` master repo? BTW the changes look good

**Comment by lxbsz** [comment]
[...]
> > Or if the job is scheduled with the `rhel/Centos` OS the test will pass too. Only will it fail in `Ubuntu`.
> 
> okay but when running full suite, it will show failures for `Ubuntu`, so is it feasible to merge this or wait till your fix is merged in `xfstests-dev` master repo? BTW the changes look good

This failure is not related to this PR.

**Comment by dparmar18** [comment]
> [...]
> 
> > > Or if the job is scheduled with the `rhel/Centos` OS the test will pass too. Only will it fail in `Ubuntu`.
> > 
> > 
> > okay but when running full suite, it will show failures for `Ubuntu`, so is it feasible to merge this or wait till your fix is merged in `xfstests-dev` master repo? BTW the changes look good
> 
> This failure is not related to this PR.

right, i was just concerned if merging patch that fail even if not related should be done straightway(if the code looks good) or wait until the unrelated issue is patched, but anyways changes look good to me

**Review by dparmar18 — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
@lxbsz This is ready for QA, should I pick it up?

**Comment by lxbsz** [comment]
> @lxbsz This is ready for QA, should I pick it up?

Sure, please. Thanks.

**Comment by lxbsz** [comment]
> > @lxbsz This is ready for QA, should I pick it up?
> 
> Sure, please. Thanks.

@rishabh-d-dave BTW, What's the status for the tests ? The `xfstests-dev` fixing have been applied already. Thanks

**Comment by lxbsz** [comment]
jenkins test make check arm64

**Comment by lxbsz** [comment]
I will run all the relevant tests myself today.

**Comment by rishabh-d-dave** [comment]
> > > @lxbsz This is ready for QA, should I pick it up?
> > 
> > 
> > Sure, please. Thanks.
> 
> @rishabh-d-dave BTW, What's the status for the tests ? The `xfstests-dev` fixing have been applied already. Thanks

I initiated testing a couple of times but there were lots of infra-related failures in both runs. I'll initiate it a third time today.

**Comment by lxbsz** [comment]
> > > > @lxbsz This is ready for QA, should I pick it up?
> > > 
> > > 
> > > Sure, please. Thanks.
> > 
> > 
> > @rishabh-d-dave BTW, What's the status for the tests ? The `xfstests-dev` fixing have been applied already. Thanks
> 
> I initiated testing a couple of times but there were lots of infra-related failures in both runs. I'll initiate it a third time today.

Sure, thanks.

**Comment by lxbsz** [comment]
@rishabh-d-dave @mchangir 

I have run the and passed:

https://pulpito.ceph.com/xiubli-2023-06-09_04:16:04-fs:functional-wip-lxb-fscrypt-20230607-0901-distro-default-smithi/
https://pulpito.ceph.com/xiubli-2023-06-09_03:59:14-fs:fscrypt-wip-lxb-fscrypt-20230607-0901-distro-default-smithi/

If no objection I will merge this and the kclient fscrypt patches need to run these tests, we need to push the kclient fscrypt patches further.

Thanks

**Review by mchangir — APPROVED** [review]
LGTM

**Comment by lxbsz** [comment]
Talked with @rishabh-d-dave, this is fine with him. Since we need to push the fscrypt patches in kclient and the tests passed. Merge it. Thanks!

**Pull request merged**
